src/instamatic/calibrate/calibrate_stage_rotation.py

```diff
@@ -0,0 +1,395 @@
+    speed_range: `Optional[Sequence[Union[float, int]]]`
+        Spead range to measure. Default: range(1, 13, 1) or range(.1, 1.1, .1).
+    outdir: `str` or None
+        Directory where the final calibration file will be saved.
+
+    return:
+        instance of `CalibStageRotation` class with conversion methods
+    """
+
+    alpha_spans = np.array(alpha_spans or np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
+    alternating_ones = np.ones(len(alpha_spans)) * (-1) ** np.arange(len(alpha_spans))
+    alpha_targets = np.cumsum(alpha_spans * alternating_ones)
+
+    calib_points: List[SpanSpeedTime] = []
+    starting_stage_alpha = ctrl.stage.a
+    starting_stage_speed = ctrl.stage.get_rotation_speed()
+    ctrl.cam.block()
+    try:
+        try:
+            ctrl.stage.set_rotation_speed(12)
+            assert ctrl.stage.get_rotation_speed() == 12
+        except AssertionError:  # or any other raised if speed can't be set
+            speed_range_default = np.linspace(0.01, 0.2, num=20)
+            speed_options = FEI_ROTATION_SPEED_OPTIONS
+        else:
+            speed_range_default = np.arange(1, 13, step=1)
+            speed_options = JEOL_ROTATION_SPEED_OPTIONS
+        speed_range = speed_range or speed_range_default
+
+        total = len(speed_range) * len(alpha_spans)
+        log(f'Starting rotation speed calibration based on {total} points.')
+
+        with tqdm(total=total) as progress_bar:
+            for speed in speed_range:
+                with ctrl.stage.rotation_speed(speed=float(speed)):
+                    ctrl.stage.a = 0.0
+                    for at, as_ in zip(alpha_targets, alpha_spans):
+                        with timer() as t:
+                            ctrl.stage.a = float(at)
+                        calib_points.append(SpanSpeedTime(as_, speed, t()))
+                        progress_bar.update(1)
+    finally:
+        ctrl.stage.set(a=starting_stage_alpha)
+        ctrl.stage.set_rotation_speed(starting_stage_speed)
+        ctrl.cam.unblock()
+
+    calib_points_array = np.array(calib_points).T
+    all_spans_speeds = calib_points_array[:2]
+    all_times = calib_points_array[2]
+    f = CalibStageRotation.curve_fit_model
+    p = curve_fit(f, all_spans_speeds, ydata=all_times, p0=[1, 0, 0])
+    (alpha_pace, alpha_windup, delay), p_cov = p  # noqa - this unpacking is OK
+    alpha_pace_u, alpha_windup_u, delay_u = np.sqrt(np.diag(p_cov))
+
+    log('Stage rotation speed calibration fit complete:')
+    log(f'alpha_pace   = {alpha_pace:12.6f} +/- {alpha_pace_u:12.6f} s / deg')
+    log(f'alpha_windup = {alpha_windup:12.6f} +/- {alpha_windup_u:12.6f} s')
+    log(f'delay        = {delay:12.6f} +/- {delay_u:12.6f} s')
+    log('model time   = (alpha_pace * alpha_span + alpha_windup) / speed + delay')
+
+    c = CalibStageRotation(*p[0], speed_options=speed_options)
+    c.to_file(None if outdir is None else Path(outdir) / CALIB_STAGE_ROTATION)
```

Comment:
See comment on `c.to_file()`

```suggestion
    c.to_file(outdir=outdir)
```

Reply:
Agreed and applied, see joint comment below.